pkg/services/accesscontrol/resourcepermissions/api_adapter.go

```diff
@@ -0,0 +1,147 @@
+	}
+
+	resourcePermName := a.buildResourcePermissionName(resourceID)
+
+	resourcePermResource := dynamicClient.Resource(iamv0.ResourcePermissionInfo.GroupVersionResource()).Namespace(namespace)
+	resourcePerm, err := resourcePermResource.Get(ctx, resourcePermName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return getResourcePermissionsResponse{}, nil
+		}
+		return nil, fmt.Errorf("failed to get resource permission from k8s: %w", err)
+	}
+
+	return a.convertK8sResourcePermissionToDTO(resourcePerm)
+}
+
+func (a *api) convertK8sResourcePermissionToDTO(resourcePerm *unstructured.Unstructured) (getResourcePermissionsResponse, error) {
+	permissions, found, err := unstructured.NestedSlice(resourcePerm.Object, "spec", "permissions")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get permissions from spec: %w", err)
+	}
+	if !found {
+		return getResourcePermissionsResponse{}, nil
+	}
+
+	dto := make(getResourcePermissionsResponse, 0, len(permissions))
+
+	for _, permRaw := range permissions {
+		permMap, ok := permRaw.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		kind, _, _ := unstructured.NestedString(permMap, "kind")
+		name, _, _ := unstructured.NestedString(permMap, "name")
+		verb, _, _ := unstructured.NestedString(permMap, "verb")
+
+		if name == "" || verb == "" {
+			continue
+		}
+
+		actions, exists := a.service.options.PermissionsToActions[verb]
+		if !exists {
+			log.New("resource-permissions-api").Warn(
+				"Permission verb not found in PermissionsToActions map",
+				"verb", verb,
+				"resource", a.service.options.Resource,
+				"availablePermissions", fmt.Sprintf("%v", getMapKeys(a.service.options.PermissionsToActions)),
+			)
+			actions = []string{}
+		}
+
+		permission := verb
+
+		permDTO := resourcePermissionDTO{
+			Permission:  permission,
+			Actions:     actions,
+			IsManaged:   true,
+			IsInherited: false,
+		}
+
+		switch iamv0.ResourcePermissionSpecPermissionKind(kind) {
+		case iamv0.ResourcePermissionSpecPermissionKindUser, iamv0.ResourcePermissionSpecPermissionKindServiceAccount:
+			userDetails, err := a.service.userService.GetByUID(context.Background(), &user.GetUserByUIDQuery{UID: name})
+			if err == nil {
+				permDTO.UserID = userDetails.ID
+				permDTO.UserUID = userDetails.UID
+				permDTO.UserLogin = userDetails.Login
+				permDTO.UserAvatarUrl = dtos.GetGravatarUrl(a.cfg, userDetails.Email)
+				permDTO.IsServiceAccount = userDetails.IsServiceAccount
+			}
+		case iamv0.ResourcePermissionSpecPermissionKindTeam:
+			permDTO.TeamUID = name
+			permDTO.Team = name
+		case iamv0.ResourcePermissionSpecPermissionKindBasicRole:
+			permDTO.BuiltInRole = name
+		}
+
+		permDTO.RoleName = fmt.Sprintf("managed:%s:%s:permissions", a.service.options.Resource, name)
+
+		dto = append(dto, permDTO)
+	}
+
+	return dto, nil
+}
+
+func (a *api) getAPIGroup() string {
+	switch a.service.options.Resource {
+	case "folders":
+		return "folder.grafana.app"
```

Comment:
could we use the constants instead from the apps folder? (APIGroup in apps/folder... / apps/dashboard/..)